src/profile-logic/marker-data.js

```diff
@@ -382,58 +417,49 @@ export function correlateIPCMarkers(threads: Thread[]): IPCMarkerCorrelations {
     }
     if (typeof thread.tid === 'number') {
       const tid: number = thread.tid;
-      threadNames.set(thread.tid, getFriendlyThreadName(threads, thread));
+      threadNames.set(tid, getFriendlyThreadName(threads, thread));
 
       for (let index = 0; index < thread.markers.length; index++) {
         const data = thread.markers.data[index];
-        if (!data || data.type !== 'IPC' || data.direction !== 'receiving') {
+        if (!data || data.type !== 'IPC') {
           continue;
         }
-        const key = makeMarkerID(data.otherPid, thread.pid, data);
-        if (dstMarkers.has(key)) {
-          console.warn('Duplicate IPC marker found for key', key);
+        const key = makeMarkerID(thread, data);
+        if (!markersByKey.has(key)) {
+          markersByKey.set(key, new Array(5).fill(undefined));
+        }
+
+        const currMarkers = ((markersByKey.get(key): any): Array<Object>);
```

Comment:
Instead of coercing to any, please use the `ensureExists()` function, or handle the `undefined` result.

Reply:
Cool, I didn't know `ensureExists()`, well... existed! Fixed.